go/enclave/rollupchain/rollup_chain.go

```diff
@@ -245,10 +247,30 @@ func (rc *RollupChain) handleGenesisRollup(b *types.Block, rollups []*obscurocor
 		}
 		rc.storage.SaveNewHead(&bs, genesis, nil)
 		s := rc.storage.GenesisStateDB()
+
+		faucetAddress := gethcommon.HexToAddress(faucetAddressHex)
+
+		// We preallocate funds to the faucet address.
+		s.SetBalance(faucetAddress, big.NewInt(faucetPrealloc))
```

Comment:
can you also try with AddBalance instead of SetBalance?

Reply:
Error was elsewhere.